Code/Legacy/CrySystem/DebugCallStack.cpp

```diff
@@ -179,7 +180,7 @@ void MarkThisThreadForDebugging(const char* name)
 
 void UnmarkThisThreadFromDebugging()
 {
-    WriteLock lock(g_lockThreadDumpList);
+    AZStd::lock_guard<AZStd::spin_mutex> lock(g_lockThreadDumpList);
```

Comment:
This can be changed to be
```suggestion
    AZStd::scoped_lock lock(g_lockThreadDumpList);
```